### PR TITLE
Add setup.py and Travis testing

### PR DESCRIPTION
**Pull request by blink1073**
Tracks https://github.com/jupyterlab/jupyterlab/pull/1876.  After this is merged I will cut a release and use it in that PR in `setup_requires`.

### CODE REVIEW

**Comment by blink1073** [comment]
@gnestor this is ready for review.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from __future__ import print_function
+from setuptools import setup
+from jupyter_packaging import __version__, create_cmdclass
```
Comment (gnestor) [inline setup.py]: Eating our own dogfood? 👍

File: jupyter_packaging.py [hunk 2]
```diff
@@ -147,15 +143,82 @@ def get_outputs(self):
...
+    build_dir: str, optional
+        The target build directory.  If this and source_dir are given,
+        the JavaScript will only be build if necessary.
+    source_dir: str, optional
+        The source code directory.
+    build_cmd: str, optional
+        The npm command to build assets to the build_dir.
+    """
+
+    class NPM(BaseCommand):
```
Comment (gnestor) [inline jupyter_packaging.py]: Nice!

File: jupyter_packaging.py [hunk 3]
```diff
@@ -42,6 +46,13 @@ def list2cmdline(cmd_list):
     os.environ.get('PATH', os.defpath),
 ])
 
+if "--skip-npm" in sys.argv:
+    print("Skipping install of webtools as requested.")
```
Comment (gnestor) [inline jupyter_packaging.py]: "webtools"? How about "Skipping `npm install`..."

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+language: python
```
Comment (gnestor) [inline .travis.yml]: 🙏

**Comment by blink1073** [comment]
https://pypi.python.org/pypi/jupyter-packaging

**Pull request merged**